File: algorithms/sort/selection/C/selection.c
#include <stdio.h>
#include <stdlib.h>

void selectionSort(int *arr, int n) {
    for (int i = 0; i < n; i++) {
        int min_idx = i;
        for (int j = i+1; j < n; j++) {
            if (arr[j] < arr[min_idx])
                min_idx = j;
        }
        if (min_idx != i) {
            int temp = arr[i];
            arr[i] = arr[min_idx];
            arr[min_idx] = temp;
        }
    }
}

int main() {
    int n; int *arr;
    scanf("%d", &n);

    arr = (int*) malloc((n + 1) * sizeof(int));
    for (int i = 0; i < n; i++) {
        int temp = 0;
        scanf("%d", &temp);
        arr[i] = temp;
    }

    selectionSort(arr, n);

    for (int i=0; i<n; i++)
        printf("%d ", arr[i]);

    return 0;
}

File: algorithms/sort/selection/python/selection.py
"""
>>> a = [1, 2, 3, 4, 5]
>>> selection_sort(a)
>>> a
[1, 2, 3, 4, 5]
>>> a = [4, 2, 5, 1, 3]
>>> selection_sort(a)
>>> a
[1, 2, 3, 4, 5]
"""


def selection_sort(a):
    length = len(a)
    for i in range(length):
        min_idx = i
        for j in range(i+1, length):
            if a[j] < a[min_idx]:
                min_idx = j
        if min_idx != i:
            a[i], a[min_idx] = a[min_idx], a[i]


if __name__ == "__main__":
    import doctest
    doctest.testmod()
